tests/utils/state-locks-filter

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+
+
+class LockTrace:
+    def __init__(self, lines: list[str]):
+        self.lines = lines
+
+    def get_trace_lines(self) -> list[str]:
+        return self.lines
+    
+    def match(self, part: str) -> bool:
+        for line in self.lines:
+            if part in line:
+                return True
+        
+        return False
+    
+    def __str__(self) -> str:
+        return "".join(self.lines)
+
+    def __hash__(self):
+      return hash(str(self))
+
+    def __eq__(self, other) -> bool: 
+        if not isinstance(other, LockTrace):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+
+        return hash(str(self)) == hash(str(other))
+
+class Lock:
+    def __init__(self, lines: list[str]):
+        self.trace = LockTrace(lines[1:])
+        self.header = lines[0]
+        self.held_time = 0
+        self.wait_time = 0
+        self._calc_held_ms(self.header)
+        self._calc_wait_ms(self.header)
+
+    def _calc_held_ms(self, line: str) -> int:
+        match = re.search('.*held: (.+?) ms.*', line)
+        if match:
+            self.held_time = int(match.group(1))
+
+    def _calc_wait_ms(self, line: str) -> int:
+        match = re.search('.*wait (.+?) ms.*', line)
+        if match:
+            self.wait_time = int(match.group(1))
+
+    def get_held_time(self) -> int:
+        return self.held_time
+
+    def get_wait_time(self) -> int:
+        return self.wait_time
+
+    def get_trace(self) -> LockTrace:
+        return self.trace
+
+class LocksGroup:
```

Comment:
the classes need a little bit of a doc comments, I'm not quite sure what a LocksGroup or a Lock is in the of the script,  is Lock a lock attempt?